include/alpaka/block/shared/st/detail/BlockSharedMemStMemberImpl.hpp

```diff
@@ -142,11 +142,10 @@ namespace alpaka
             template<typename T>
             std::uint32_t varChunkEnd(uint32_t byteOffset) const
             {
-                static_assert(
-                    core::vectorization::defaultAlignment >= alignof(T),
-                    "Unable to get block shared static memory for types with alignment higher than defaultAlignment!");
-                constexpr std::uint32_t align = static_cast<std::uint32_t>(std::max(TMinDataAlignBytes, alignof(T)));
-                return (byteOffset / align + (byteOffset % align > 0u)) * align + static_cast<uint32_t>(sizeof(T));
+                size_t const ptr = reinterpret_cast<size_t>(m_mem + byteOffset);
+                constexpr size_t align = std::max(TMinDataAlignBytes, alignof(T));
+                size_t const newPtrAdress = ((ptr + align - 1u) / align) * align + sizeof(T);
+                return static_cast<uint32_t>(newPtrAdress - reinterpret_cast<size_t>(m_mem));
```

Comment:
This smells a bit like `std::align`: https://en.cppreference.com/w/cpp/memory/align

But your implementation looks good as well.

Reply:
You are right but the special requirement is that we can get the address from other threads too. There is no guarantee how `std::align` is implemented.

Reply:
Good point! Leave it then as it is.